llvm/lib/Transforms/Utils/SimplifyCFG.cpp

```diff
@@ -6631,6 +6630,27 @@ static bool SwitchToLookupTable(SwitchInst *SI, IRBuilder<> &Builder,
 
   BranchInst *RangeCheckBranch = nullptr;
 
+  // Grow the table to cover all possible index values to avoid the range check.
+  // It will use the default result to fill in the table hole later, so make
+  // sure it exist.
+  if (UseSwitchConditionAsTableIndex && HasDefaultResults) {
```

Comment:
Should this check HasDefaultResults or !DefaultIsReachable?

Reply:
To clarify: If the default is already not reachable, then there is no point in increasing the table size in the first place, as no branch will be generated.

Reply:
hi @nikic , I think your intention  is  `DefaultIsReachable && HasDefaultResults` ? 
If so, I think it is more accurate to match above clarification.

Reply:
Thank you for reconfirming